src/main/java/frc/robot/subsystems/CDSSubsystem.java

```diff
@@ -235,24 +259,73 @@ public int getBallCount() {
     return ballCount;
   }
 
-  public boolean getReady() {
-    return isReady;
+  public boolean getMissedSensor(){
+    return missed;
+  }
+
+  public void setMissedSensor(boolean missedBall){
+    missed = missedBall;
+  }
+
+  public boolean getMissedColor(){
+    return missedColor;
+  }
+
+  public void setMissedColor(boolean colorMissed){
+    missedColor = colorMissed;
   }
 
-  public void setReady(boolean status) {
-    isReady = status;
+  public String getState() {
+    return state.toString();
   }
 
-  public void periodic() {}
-
-  private void instantiateDebugTab() {
-    CDSTab = Shuffleboard.getTab("CDS Tab");
-    CDSBeltControllerDirection =
-        CDSTab.add("CDS Belt Direction", "Not Running")
-            .withPosition(2, 0)
-            .withWidget(BuiltInWidgets.kToggleSwitch)
-            .getEntry();
-    CDSWheelControllerSpeed = CDSTab.add("CDS Wheel speed", 0).withPosition(3, 0).getEntry();
-    CDSBeltControllerSpeed = CDSTab.add("CDS Belt speed", 0).withPosition(4, 0).getEntry();
+  public void periodic() {
+    int ballCount = getBallCount();
+    String ballColor = senseColor();
+
+    int currentOpenSensor = getNextOpenSensor();
+    
+    switch (state) {
+      case IDLE:
+        nextOpenSensor = -1;
+        msCurrent = 0;
+
+        if (ballCount > 2 || (getMissedColor() && getMissedSensor())) {
+          state = MangementState.EJECT;
+        } 
+        
+         if (ballCount < 3 && currentOpenSensor != -1 && getMissedSensor()) {
```

Comment:
We shouldn't need the missed sensor stuff anymore since this will always be running.